### PR TITLE
Implements a basic skill system + adds surgery skill 

### PR DESCRIPTION
**Pull request by silicons**
# THIS IS NOT BAYSTATION SKILLS.

## About The Pull Request

Adds a basic skill system, supporting binary, enum, and the most useful, numerical skills.
Adds surgery skill, providing 40% speedup at 100 skill.
Each successful surgery step provides 0.25 skill.
Medical staff gets 35/100 skill at the start of the game automatically (around 14% speedup), and gains skill 1.2x faster.

## Why It's Good For The Game

A skill system's been talked about for a while to make jobs better at.. well, their job.
This is a cross between something like hardcoded job skills and /tg/skills where you gain skill for doing your job.

Let me make it clear that this is not baystation skills and I resent those kinds of skill ("you can not do this because you aren't <x>") systems very much.
You have my reassurance that this isn't going to be a slippery slope towards said baystation skill system being normalized in gameplay.

## Changelog
:cl:
add: Skills have been added.
add: Surgery is now faster if you are skilled. Doctors are a bit trained roundstart and train this faster, otherwise, just improve it by doing surgery steps.
/:cl:

### CODE REVIEW

**Comment by Arturlang** [comment]
Gamer skill when

**Comment by Enzo-Leon** [comment]
Hell yeah I been looking forward to this <3!

**Comment by BobJogaIsForkinBanned** [comment]
ok i do like this very much

File: code/modules/jobs/job_types/geneticist.dm [hunk 1]
```diff
@@ -19,6 +19,8 @@
 	display_order = JOB_DISPLAY_ORDER_GENETICIST
 	threat = 1.5
 
+	starting_skills = list(/datum/skill/numerical/surgery = STARTING_SKILL_SURGERY_MEDICAL)
```
Comment (Linzolle) [inline code/modules/jobs/job_types/geneticist.dm:22]: why doesn't the geneticist get the affinity when the other non-surgeon jobs also get it?

Comment (Poojawa) [reply code/modules/jobs/job_types/geneticist.dm:22]: because they're a science department employee :^)

Comment (silicons) [reply code/modules/jobs/job_types/geneticist.dm:22]: @Linzolle the real answer is i forgot

the answer i'd rather use is what @Poojawa said
they're like
not as Medical as the other Medical Dudes 

i mean i'll change it if you want :^)

Comment (Linzolle) [reply code/modules/jobs/job_types/geneticist.dm:22]: eh i dont really care was just wondering why lol

**Review by Linzolle — APPROVED** [review]

**Pull request merged**